packages/eas-cli/src/commands/update/index.ts

```diff
@@ -584,6 +571,59 @@ export default class UpdatePublish extends EasCommand {
   }
 }
 
+export async function getUpdatesToRepublishInteractiveAsync(
+  projectId: string,
+  branchName: string,
+  platformFlag: string,
+  cumulativeUpdates: Exclude<
+    Exclude<ViewBranchUpdatesQuery['app'], null | undefined>['byId']['updateBranchByName'],
+    null | undefined
+  >['updates'] = [],
+  offset: number = 0
+): Promise<any> {
+  const fetchMoreValue = '_fetchMore';
+
+  const { updates } = await ensureBranchExistsAsync({
+    appId: projectId,
+    name: branchName,
+    offset,
+  });
+  cumulativeUpdates = [...cumulativeUpdates, ...updates];
+  const updateGroups = uniqBy(cumulativeUpdates, u => u.group)
+    .filter(update => {
+      // Only show groups that have updates on the specified platform(s).
+      return platformFlag === 'all' || update.platform === platformFlag;
+    })
+    .map(update => ({
+      title: formatUpdateTitle(update),
+      value: update.group,
+    }));
+  if (!updateGroups.length) {
+    throw new Error(
+      `There are no updates on branch "${branchName}" published on the platform(s) ${platformFlag}. Did you mean to publish a new update instead?`
+    );
+  }
+
+  if (updates.length && updates.length === getViewBranchUpdatesQueryUpdateLimit()) {
```

Comment:
updates.length will always be defined
```suggestion
  if (updates.length === getViewBranchUpdatesQueryUpdateLimit()) {
```

Reply:
i think this makes sense. i was trying to avoid the case where both items were 0... but that should never happen regardless.

Reply:
Gotcha... the pageSize+1 suggestion I left in earlier comments should avoid this case too.